codingstandards/listproperties.md

```diff
@@ -0,0 +1,128 @@
+# List Properties
+
+Many types, especially data structures, will contain properties of a list type,
+such as `IList<T>` or `IReadOnlyList<T>`.  List properties should be implemented
+under the following guidelines, especially `public` properties.
+
+- The list property itself should be read-only.
+- The list property should be initialized to a new instance when the containing data structure is constructed.
+- The list property should be of an interface type when possible.  That is, use `IList<T>` rather than `List<T>`.
+- Use the read-only collections when appropriate, such as `IReadOnlyCollection<T>`, `IReadOnlyList<T>`, and `IReadOnlyDictionary<TKey,TValue>`.
+
+## Validation
+
+As with any type, validation of the input should be performed as early as possible.
+In most cases, this can be accomplished in the constructor of the class.
+In the cases of lists, however, when a list is expected to only contain certain values,
+validation needs to occur as values are added, inserted, or set using an indexer.
+
+Here is an example of how to **not** validate the entries of a list.
+Note the list property, `Attributes`, correctly defined as a read-only property as `IList<string>`.
+
+```csharp
+public class Example
+{
+   public string Property { get; }
+
+   public IList<string> Attributes { get; } = new List<string>();
+
+   public Example(string property)
+   {
+      Property = property;
+   }
+}
+```
+
+Lets say the `Attributes` list should not contain any null or empty values.
+Before performing an operation, we may want to validate the `Attributes` property
```

Comment:
@johntkeller usually in markdown, a linebreak is seen as just that: a newline. In other markup, this is usually ignored.

In this case, you are breaking the line intentionally, no matter what the size of the container is. which could be good and bad as it would lead to something like this in smaller resolutions:

**Your breaks**
>Lorem Ipsum is simply dummy text of the printing and _break_
>typesetting_ industry. Lorem Ipsum has been the industry's _break_
>standard dummy text ever since the 1500s, when an unknown _break_
>printer took a galley of type and scrambled it to make a type _break_
>specimen book.

**Smaller Resolution**
>Lorem Ipsum is simply dummy text of the _resolution limit_
>printing and _break_
>typesetting_ industry. Lorem Ipsum has been _resolution limit_
>the industry's _break_
>standard dummy text ever since the 1500s, _resolution limit_
>when an unknown _break_
>printer took a galley of type and scrambled it to _resolution limit_
>make a type _break_
>specimen book.

Regardless, not a big deal, but thought I'd share my 0.02 :)